docs/guides/features/ipsec-guide.md

```diff
@@ -0,0 +1,135 @@
+[IPsec Recipe](https://github.com/ipdk-io/ipsec-recipe), which provides
+strongSwan plugin code to configure and program control messages into the
+target.
+
+The IPsec offload feature is only supported on the Intel&reg; IPU E2100 target.
+
+## Feature overview
+
+The `infrap4d` process provides the gRPC server-side support for P4RT and
+gNMI messages. strongSwan acts as the gRPC client in this context.
+Alternatively, you can use other clients that implement the IKE stack. The
+client communicates using P4RT to program Security Policy Database (SPD) and
+uses gNMI to configure Security Association Database (SAD) entries, which
+includes encryption keys, re-keying etc.
+
+The [openconfig-ipsec-offload](https://github.com/ipdk-io/openconfig-public/blob/master/release/models/ipsec/openconfig-ipsec-offload.yang)
+YANG model is used to configure IPsec offload.
+
+Follow the sequence of steps listed below to enable IPsec functionality.
+
+### Load crypto package
+
+Follow the instructions in [Deploying P4 programs](/guides/es2k/deploying-p4-programs.md)
+to load the hardware FXP pipeline with the crypto package.
+
+### Configure and run infrap4d
+
+The OpenConfig portion of IPsec Offload functionality is enabled by using fixed
+functions configuration. Follow the instructions in [Running infrap4d](/guides/es2k/running-infrap4d.md)
+to prepare system with generated TDI.json and context.json file references.
+
+The /usr/share/stratum/es2k/es2k_skip_p4.conf file must include the fixed
+function configuration reference.
+
+```json
+"fixed_functions" : [
+  {
+   "name": "crypto",
+   "tdi": "/tmp/fixed/tdi.json",
+   "ctx": "/tmp/fixed/crypto-mgr-ctx.json"
+  }
+],
+```
+
+With this configuration updated, you can start infrap4d.
+
+### Compile P4 program
+
+Compile the ipsec-offload program according to the instructions in
+[Compiling P4 programs](/guides/es2k/compiling-p4-programs.md)
+to generate crypto P4 artifacts for programming the pipelines.
+
+### Configure strongSwan
+
+Follow the instructions in the [strongSwan documentation](https://docs.strongswan.org/docs/5.9/index.html)
+to configure host for the use-case selected. This includes details such as
+certificate location, IPs, lifetime thresholds etc.
+
+### Start IPsec
+
+With the strongSwan application configured, starting IPsec
+(see [ipsec-recipe](https://github.com/ipdk-io/ipsec-recipe) for details) will 
+initiate the pipeline, program the SPD rules as per the P4 program, and
+configure/re-configure SAD entries based on negotiated encryption parameters
+between local and peer system.
+
+## OpenConfig and gNMI messages
+
+This section provides detailed information on OpenConfig model and gNMI
+messages with the expected format. The strongSwan plugin has the following
+details encoded. This is only provided for clarity and information purposes.
+
+### The Config SAD message
+
+The P4 Control Plane provides SET and DELETE API calls service to the 
```

Comment:
```suggestion
The SET and DELETE API calls . . .
```
You're describing the individual messages. They (not P4 Control Plane) are your topic.

"The GET and DELETE API calls service to the IPsec-offload Config SAD message at . . ." doesn't make sense.

Perhaps something along the lines of:

"The SET and DELETE API calls are used to configure the Security Association Database. See the IPsec-offload [Config SAD message](https://github.com/ipdk-io/openconfig-public/blob/master/release/models/ipsec/openconfig-ipsec-offload.yang#L39-L185) ..."